libs/log/tm_logger.go

```diff
@@ -52,15 +53,38 @@ func NewTMLoggerWithColorFn(w io.Writer, colorFn func(keyvals ...interface{}) te
 // Info logs a message at level Info.
 func (l *tmLogger) Info(msg string, keyvals ...interface{}) {
 	lWithLevel := kitlevel.Info(l.srcLogger)
+
+	// This value will be marshalled accordingly to the
```

Comment:
maybe we should move this to https://github.com/tendermint/tendermint/blob/09cf0bcb012d37de33b41425d82a77bbb31ed1b5/libs/log/tmfmt_logger.go#L107

that way we'll eliminate duplication

Reply:
Does tmfmt_logger wrap tm_logger? Maybe I misunderstand how these loggers are used.

Reply:
I believe `tm_logger` wraps `tmfmt_logger`

Reply:
Wouldn't we want to convert this to hexbytes here in tm_logger, because  and it's unnecessary for srcLogger to handle the conversion/have knowledge of the hexByte conversion? Seems like if we switched out the implementation of srcLogger at some future date, this may cause issues in the future (or failure to move the hexbytes conversion into this new implementation), right? 

What is the benefit of moving this to the other logger?

Reply:
I think I found where `tm_logger` wraps `tmfmt_logger`: https://github.com/tendermint/tendermint/blob/master/libs/log/tm_logger.go#L43

I'm not familiar with the intended architecture design of the loggers, but @laniehei brings up some good questions about how future changes to the logging design could cause unintended regressions in this specialized formatting. How could we protect against this type of regression?

Reply:
> What is the benefit of moving this to the other logger?

de duplication of the code. we won't have to repeat the same logic 3 times (Info, Debug, Error).

> Wouldn't we want to convert this to hexbytes here in tm_logger, because and it's unnecessary for srcLogger to handle the conversion/have knowledge of the hexByte conversion?

no. if you look at the tmfmt_logger, you will see that it already has deep knowledge of the structure and does formatting (what we're trying to do here as well).